llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -901,6 +916,31 @@ bool modifyToS20(InstrNode Start, MachineRegisterInfo &MRI, MachineIRBuilder &B,
     Helper.tryCombineCopy(*StartNodeMI);
     return true;
   }
+  case TargetOpcode::G_INTRINSIC: {
+    if (isS20Intrinsic(*StartNodeMI, MRI)) {
+      Register ExtractDstReg = StartNodeMI->getOperand(0).getReg();
+      // Note: Operand 1 is the ID of the intrinsic
+      const Register SrcReg0 = StartNodeMI->getOperand(2).getReg();
+      const Register SrcReg1 = StartNodeMI->getOperand(3).getReg();
+      const Register SignReg = StartNodeMI->getOperand(4).getReg();
+
+      auto SignVal = getIConstantVRegSExtVal(SignReg, MRI);
+      assert(SignVal.has_value() && "Expected SignVal to be constant");
+      // Erase the original instruction and set the type for the destination
+      // register
+      Helper.eraseInst(*StartNodeMI);
+      MRI.setType(ExtractDstReg, S20);
+      const unsigned Opcode = SignVal.value()
+                                  ? AIE2::G_AIE_SEXT_EXTRACT_VECTOR_ELT
+                                  : AIE2::G_AIE_ZEXT_EXTRACT_VECTOR_ELT;
+      StartNodeMI =
+          B.buildInstr(Opcode, {ExtractDstReg}, {SrcReg0, SrcReg1}).getInstr();
```

Comment:
Check : How do we make sure that the newly created instruction is in right order. By order I mean is inserted between expected set of instruction